src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/RemoveAzureApiManagementIdentityProvider.cs

```diff
@@ -0,0 +1,69 @@
+﻿//  
+// Copyright (c) Microsoft.  All rights reserved.
+// 
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+// 
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+namespace Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Commands
+{
+    using Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Models;
+    using Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Properties;
+    using System;
+    using System.Globalization;
+    using System.Management.Automation;
+
+    [Cmdlet(VerbsCommon.Remove, Constants.ApiManagementIdentityProvider, SupportsShouldProcess = true)]
+    [OutputType(typeof(bool))]
+    public class RemoveAzureApiManagementIdentityProvider : AzureApiManagementCmdletBase
+    {
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = true,
+            HelpMessage = "Instance of PsApiManagementContext. This parameter is required.")]
+        [ValidateNotNullOrEmpty]
+        public PsApiManagementContext Context { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = true,
+            HelpMessage = "Identifier of existing identity provider. This parameter is required.")]
+        [ValidateNotNullOrEmpty]
+        public PsApiManagementIdentityProviderType Type { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "If specified will write true in case operation succeeds. This parameter is optional. Default value is false.")]
+        public SwitchParameter PassThru { get; set; }
+
+        public override void ExecuteApiManagementCmdlet()
+        {
+            var actionDescription = string.Format(CultureInfo.CurrentCulture, Resources.IdentityProviderRemoveDescription, Type.ToString("g"));
+            var actionWarning = string.Format(CultureInfo.CurrentCulture, Resources.IdentityProviderRemoveWarning, Type.ToString("g"));
+
+            // Do nothing if force is not specified and user cancelled the operation
+            if (!ShouldProcess(
+                    actionDescription,
+                    actionWarning,
+                    Resources.ShouldProcessCaption))
+            {
+                return;
+            }
+
+            Client.IdentityProviderRemove(Context, Type.ToString("g"));
```

Comment:
This is a bit of a confusing construction, please just use a positive check for ShouldProcess and place the implementation inside the 'then' block.